### PR TITLE
Add additional methods to support gfx gl backend improvements

### PR DESCRIPTION
**Pull request by kyren**
You will need this for the gfx-hal webgl PR once [this pr](https://github.com/gfx-rs/gfx/pull/2776) lands.

### CODE REVIEW

File: src/native.rs [hunk 1]
```diff
@@ -775,6 +775,14 @@ impl super::Context for Context {
         gl.GetAttribLocation(program, name.as_ptr() as *const i8) as i32
     }
 
+    unsafe fn get_sync_status(&self, fence: Self::Fence) -> u32 {
+        let gl = &self.raw;
+        let mut len = 0;
+        let mut values = [UNSIGNALED as i32];
+        gl.GetSynciv(fence, SYNC_STATUS, values.len() as i32, &mut len, values.as_mut_ptr());
```
Comment (grovesNL) [inline src/native.rs]: I think `bufSize` might be in bytes here (instead of `values.len()`)

Comment (kyren) [reply src/native.rs]: https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glGetSync.xhtml

It says "up to bufSize integers", but you might be right... you would think that the docs would be more clear here.

Comment (grovesNL) [reply src/native.rs]: This is a bit strange, in the OpenGL Programming Guide it says:

> bufSize is the size in bytes of the buffer whose address is given in values

I guess we can merge this for now and get clarification afterwards.

Comment (grovesNL) [reply src/native.rs]: Alright I asked around a bit (https://community.khronos.org/t/clarification-about-glgetsynciv-bufsize-parameter/104062 and in Discord), so we can fix this later if we guessed wrong

Comment (kyren) [reply src/native.rs]: If there's no clear answer, I have a cheeky suggestion: we could always pass an array of 4 u32s and a bufSize of 4, then just return the first entry 😝

**Pull request merged**